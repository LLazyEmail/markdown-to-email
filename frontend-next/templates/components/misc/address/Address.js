import React from 'react';

const Address = () => (
  <div>
    <strong>Our mailing address is:</strong>
    <br />
    PO Box 2206, Edwards CO, 81632, U.S.A.
  </div>
);

export default Address;
