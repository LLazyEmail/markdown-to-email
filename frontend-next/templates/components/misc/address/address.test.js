import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Address from './Address';

const _reactDom = require('react-dom');

const _reactDom2 = _interopRequireDefault(_reactDom);

const _index = require('./index');

const _index2 = _interopRequireDefault(_index);

const _react = require('react');

const _react2 = _interopRequireDefault(_react);

const _react3 = require('@testing-library/react');

const _reactTestRenderer = require('react-test-renderer');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

configure({ adapter: new Adapter() });

afterEach(() => {
  (0, _react3.cleanup)();
});

describe('address/index component', () => {

    it('<Address > renders with correct length 1 div element', () => {
        const wrapper = shallow(<Address />);
        expect(wrapper).toHaveLength(1);
    });

    it('<Address > element renders correct text', () => {
        const wrapper = shallow(<Address />);
        expect(wrapper.text()).toEqual('Our mailing address is:PO Box 2206, Edwards CO, 81632, U.S.A.');
    });

    test('renders without crashing', () => {
        const div = document.createElement('div');
        _reactDom2.default.render(_react2.default.createElement(_index2.default, null), div);
        _reactDom2.default.unmountComponentAtNode(div);
    });

    test('renders correctly', () => {
        const tree = (0, _reactTestRenderer.create)(_react2.default.createElement(_index2.default, null));
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
