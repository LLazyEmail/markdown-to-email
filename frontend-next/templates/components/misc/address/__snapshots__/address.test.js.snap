// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`address/index component renders correctly 1`] = `
<div>
  <strong>
    Our mailing address is:
  </strong>
  <br />
  PO Box 2206, Edwards CO, 81632, U.S.A.
</div>
`;
